src/MusicTheory/Duration.elm

```diff
@@ -0,0 +1,37 @@
+module MusicTheory.Duration exposing
+    ( Division(..)
+    , Duration(..)
+    , Grouping(..)
+    , TiedOrSeparate(..)
+    )
+
+
+type Grouping
+    = Single Duration
+    | Duplet Duration Duration
+    | Triplet Duration Duration Duration
+    | Quadruplet Duration Duration Duration Duration
+    | Quintuplet Duration Duration Duration Duration Duration
+
+
+type Division
+    = Whole
+    | Half
+    | Quarter
+    | Eighth
+    | Sixteenth
+    | ThirtySecond
+    | SixtyFourth
+    | OneHundredTwentyEighth
+
+
+type Duration
+    = Normal Division TiedOrSeparate
+    | Dotted Division TiedOrSeparate
+    | DoubleDotted Division TiedOrSeparate
+    | TripleDotted Division TiedOrSeparate
+
+
+type TiedOrSeparate
```

Comment:
Since most notes are probably not tied, I think it would be good to not have to specify this in the constructor and default to separate. Then provide a function to add a tie e.g. `duration |> tied`.

Reply:
Yeah, I think that would be a good interface.

Reply:
An interface this simple is only possible in the `Single` case. I wrote `map` functions that can be used to apply functions like `setTied`, `setPitches`, `setDotted` etc. to either all members of a tuple, or to specific members.

Reply:
looks good